src/main/java/nwoolcan/model/brewery/warehouse/QueryStockBuilder.java

```diff
@@ -0,0 +1,184 @@
+package nwoolcan.model.brewery.warehouse;
+
+import nwoolcan.model.brewery.warehouse.article.Article;
+import nwoolcan.model.utils.Quantity;
+import nwoolcan.model.utils.UnitOfMeasure;
+import nwoolcan.utils.Result;
+
+import javax.annotation.Nullable;
+import java.util.Date;
+
+/**
+ * Builder for {@link QueryStock}.
+ */
+public final class QueryStockBuilder {
+
+    private static final String UOM_ERROR = "Quantity unit of measure mismatching.\nBe sure to"
+                                          + "select an article when selecting a quantity filter"
+                                          + " and to select the same unit of measure.";
+    @Nullable
+    private Article article = null;
+    @Nullable
+    private Date expiresBefore = null;
+    @Nullable
+    private Date expiresAfter = null;
```

Comment:
this is redundant

Reply:
what are you referring to? ```@Nullable``` is not redundant at all. Try to remove it and you'll see it does not compile.

Reply:
Are you referring to the assignment to null?

Reply:
yes, i'm talking about `= null`

Reply:
I think it is more clear and I saw it also in other codes in this repo.